PlaceRouteHierFlow/placer/PlacerHyperparameters.h

```diff
@@ -21,6 +21,8 @@ class PlacerHyperparameters {
 
   bool select_in_ILP = false;
   int ilp_solver = 0; // 0 : SYMPHONY, 1 : lpsolve
+
+  std::string place_on_grid_constraints_json;
```

Comment:
@854768750 This string should be populated with the json constratins (or be empty if here are none.)

Reply:
Thanks @stevenmburns , I will see how to change the placer code.

Reply:
@854768750 Any progress here. You can check in partial progress.

Reply:
The json string is empty when I use standalone schematic2layout.py

Reply:
pytest has non empty strings. Let me use this first, although it cannot debug.

Reply:
When entering placer for G1, the string is empty.
When entering G2, the string is [G1:{}]
When entering G3, the string is [G1:{}, G2:{}]
When entering CKT_HIERARCHY, the string is [G1:{}, G2:{}, G3:{}]
Does it mean leave nodes have no constraint? Only in CKT_HIERARCHY the instance G1, G2 and G3 have some constraints?

Reply:
I thought the leaf nodes would be the constraint the same way they already are (through the primitive layout json files.)
If you want me to add them here I can do that as well. Would it be easier for you?

I think you need to turn on an environment variable (set in the particular test). If you set that, you should be able to use schematic2layout.py.